examples/david_map_reduce_weighted.py

```diff
@@ -171,6 +189,7 @@ def main(
     # Generate a coreset via uniform random sampling for comparison
     random_solver = RandomSample(coreset_size, sample_key, unique=True)
     random_coreset, _ = eqx.filter_jit(random_solver.reduce)(data)
+    random_weights = weights_optimiser.solve(data, random_coreset.coreset)
```

Comment:
Should we be computing weights here? Does it make more sense to just set the weight uniformly instead?